app/controllers/ticket.js

```diff
@@ -0,0 +1,430 @@
+const UserModel = require('../models/User')
+const HttpStatus = require('http-status-codes')
+const TicketModel = require('../models/Ticket')
+const TAGS = require('../utils/notificationTags')
+const HANDLER = require('../utils/response-helper')
+const ticketNotificationHelper = require('../utils/ticket-notif-helper')
+
+const notification = {
+  heading: '',
+  content: '',
+  tag: ''
+}
+
+module.exports = {
+
+  create: async (req, res, next) => {
+    const userId = req.user.id.toString()
+    try {
+      const ticket = new TicketModel(req.body)
+      ticket.createdBy = userId
+      ticket.history.push({ ...req.body, editedBy: userId })
+      notification.heading = 'New Support Ticket!'
+      notification.content = `${req.user.name.firstName} ${req.user.name.lastName} Creted a new Support Ticket!`
+      notification.tag = TAGS.NEW
+      await ticketNotificationHelper.addToNotificationForModerator(req, res, notification, next)
+      await ticket.save()
+      res.status(HttpStatus.CREATED).json({
+        ticket: ticket
+      })
+    } catch (error) {
+      HANDLER.handleError(res, {
+        code: error.code || HttpStatus.BAD_REQUEST,
+        ...error
+      })
+    }
+  },
+
+  getTicket: async (req, res, next) => {
+    const { user } = req.query
+    try {
+      let tickets
+      if (user === 'me') {
+        const userId = req.user.id.toString()
+        tickets = await TicketModel.find({ createdBy: userId }).lean().exec()
+      } else {
+        tickets = await TicketModel.find({}).lean().exec()
+      }
+      res.status(HttpStatus.OK).json({ tickets: tickets })
+    } catch (error) {
+      HANDLER.handleError(res, {
+        code: error.code || HttpStatus.BAD_REQUEST,
+        ...error
+      })
+    }
+  },
+
+  editTicket: async (req, res, next) => {
+    const { id } = req.params
+    const { title, content } = req.body
+    const userId = req.user.id.toString()
+    try {
+      const ticket = await TicketModel.findById(id)
+      if (!ticket) {
+        return res.status(HttpStatus.NOT_FOUND).json({ error: 'No ticket exist' })
+      }
+      if (userId !== ticket.createdBy && !req.user.isAdmin && !req.user.isTicketsModerator) {
```

Comment:
You can make a util function to check if the ticket creator is moderator or creator, because it's getting repeated